squirrel.worker/src/main/java/org/dice_research/squirrel/sink/impl/sparql/TripleBuffer.java

```diff
@@ -33,6 +35,10 @@ public void addTriple(AbstractBufferingTripleBasedSink sink, CrawleableUri uri,
     public void sendTriples(AbstractBufferingTripleBasedSink sink, CrawleableUri uri) {
         synchronized (buffer) {
             sink.sendTriples(uri, buffer);
+            if(numberOfTriples == 0) {
+                //in case of adding a triple at a later stage, numberOfTriples will be 0. So fetching it from the CrawlingActivity
+                numberOfTriples = ((CrawlingActivity) uri.getData().get(Constants.URI_CRAWLING_ACTIVITY)).getNumberOfTriples();
+            }
```

Comment:
> in case of adding a triple at a later stage, numberOfTriples will be 0

That is not true. Please have a look at the usage of the TripleBuffer class in the AbstractBufferingTripleBasedSink. The setting the internal counter `numberOfTriples` to the value of the activity does not really make sense to me.

Reply:
reverted.